src/arrayMethodReduce.js

```diff
@@ -5,7 +5,20 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let sum = initialValue;
+    let count = 0;
+
+    if (arguments.length < 2) {
+      sum = this[0];
+      count++;
+    }
+
+    while (count < this.length) {
+      sum = callback(sum, this[count], count, this);
+      count++;
+    }
```

Comment:
Use `while` loop when you don't now number of iterations, in this better to use just `for` loop